dev/com.ibm.ws.cdi.1.2.weld/src/com/ibm/ws/cdi/proxy/ProxyServicesImpl.java

```diff
@@ -53,6 +53,14 @@ public void cleanup() {
 
     @Override
     public ClassLoader getClassLoader(final Class<?> proxiedBeanType) {
+
+        if (System.getSecurityManager() != null) {
+            String className = new Exception().getStackTrace()[1].getClassName();
```

Comment:
You can do `Thread.currentThread().getStackTrace()` instead of creating an exception.